migrations/800_groups__add.sql

```diff
@@ -0,0 +1,29 @@
+-- A Work in Progress to discuss shared schema for group storage in PL
+
+CREATE TABLE IF NOT EXISTS group_configs (
+    id BIGSERIAL PRIMARY KEY,
+    course_instance_id BIGINT NOT NULL REFERENCES course_instances(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    assessment_id BIGINT REFERENCES assessments(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    type TEXT, -- what will these be? This might belong in groups rather than groups_config
+    name TEXT, -- or maybe description is accurate here? Maybe as a base for groups created from it?
+    minimum INT,
+    maximum INT,
+    date timestamp with time zone DEFAULT CURRENT_TIMESTAMP,
+    deleted_at timestamp with time zone
+);
+
+CREATE TABLE IF NOT EXISTS groups (
+    id BIGSERIAL PRIMARY KEY,
+    course_instance_id BIGINT NOT NULL REFERENCES course_instances(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    name TEXT,      -- visible name of the group
+    group_config_id BIGINT REFERENCES group_configs(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    date timestamp with time zone DEFAULT CURRENT_TIMESTAMP,
+    deleted_at timestamp with time zone
+);
+
+-- simple join table, no extra metadata - that could be stored in audit logs if needed
+CREATE TABLE IF NOT EXISTS group_users (
```

Comment:
Maybe this should be called something along the lines of `group_memberships`? `group_users` kind of makes it sound like there's a special user type created for group assessments.

Reply:
Our convention has been that join tables are named after the tables they join (users and groups).

Reply:
This was the nomenclature that Matt originally suggested. I thought people were pronouncing it "groups to users" in past meetings; I would have suggested `groups_to_users` or something. "Memberships" does sound better though. Would `group_members` be fewer letters to type?

Reply:
For some reason Dave's comment wasn't loaded at the time I commented.